devpack/src/main/java/io/neow3j/devpack/compiler/Compiler.java

```diff
@@ -2,224 +2,565 @@
-import io.neow3j.devpack.framework.Syscall;
-import io.neow3j.devpack.framework.Syscall.Syscalls;
-import io.neow3j.io.BinaryWriter;
+import io.neow3j.contract.ScriptHash;
+import io.neow3j.devpack.framework.annotations.EntryPoint;
+import io.neow3j.devpack.framework.annotations.Instruction;
+import io.neow3j.devpack.framework.annotations.Instruction.Instructions;
+import io.neow3j.devpack.framework.annotations.ManifestExtra.ManifestExtras;
+import io.neow3j.devpack.framework.annotations.ManifestFeature;
+import io.neow3j.devpack.framework.annotations.Syscall;
+import io.neow3j.devpack.framework.annotations.Syscall.Syscalls;
+import io.neow3j.model.types.ContractParameterType;
+import io.neow3j.protocol.core.methods.response.NeoGetContractState.ContractState.ContractManifest;
+import io.neow3j.protocol.core.methods.response.NeoGetContractState.ContractState.ContractManifest.ContractABI;
+import io.neow3j.protocol.core.methods.response.NeoGetContractState.ContractState.ContractManifest.ContractABI.ContractEvent;
+import io.neow3j.protocol.core.methods.response.NeoGetContractState.ContractState.ContractManifest.ContractABI.ContractMethod;
+import io.neow3j.protocol.core.methods.response.NeoGetContractState.ContractState.ContractManifest.ContractFeatures;
+import io.neow3j.protocol.core.methods.response.NeoGetContractState.ContractState.ContractManifest.ContractGroup;
+import io.neow3j.protocol.core.methods.response.NeoGetContractState.ContractState.ContractManifest.ContractPermission;
 import io.neow3j.utils.BigIntegers;
 import io.neow3j.utils.Numeric;
-import java.io.ByteArrayOutputStream;
 import java.io.IOException;
 import java.math.BigInteger;
-import java.util.Iterator;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
+import java.util.Optional;
 import org.objectweb.asm.ClassReader;
 import org.objectweb.asm.Type;
 import org.objectweb.asm.tree.AbstractInsnNode;
 import org.objectweb.asm.tree.AnnotationNode;
 import org.objectweb.asm.tree.ClassNode;
+import org.objectweb.asm.tree.IntInsnNode;
 import org.objectweb.asm.tree.LdcInsnNode;
+import org.objectweb.asm.tree.LocalVariableNode;
 import org.objectweb.asm.tree.MethodInsnNode;
 import org.objectweb.asm.tree.MethodNode;
 import org.objectweb.asm.tree.VarInsnNode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class Compiler {
 
-    public static void main(String[] args) throws Exception {
-        ClassReader reader = new ClassReader("io.neow3j.devpack.template.HelloWorldContract");
+    private static final Logger log = LoggerFactory.getLogger(Compiler.class);
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    public static final int MAX_PARAMS_COUNT = 255;
+    public static final int MAX_LOCAL_VARIABLES_COUNT = 255;
+//    public static final int MAX_STATIC_FIELDS_COUNT = 255;
+
+    private int currentNeoAddr;
```

Comment:
What's a "currentNeoAddr"? The contract address?

Reply:
That's the current opcode address of the NeoVM script being generated. I.e. the next OpCode added to the script gets this address and then the address is incremented.